src/chainstate/stacks/index/marf.rs

```diff
@@ -660,14 +685,13 @@ impl MARF {
     ///   it's a known block, the storage system isn't issueing IOErrors, _and_ it's in the same fork
     ///   as the current block
     /// The MARF _must_ be open to a valid block for this check to be evaluated.
-    pub fn check_block_hash(&mut self, bhh: &BlockHeaderHash) -> Result<(), Error> {
+    pub fn is_ancestor_block_hash(&mut self, bhh: &BlockHeaderHash) -> Result<(), Error> {
```

Comment:
Not a bug, but why not just have this method return `Result<bool, Error>` to indicate the truth of whether or not the block is an ancestor?  It feels weird to me to return an `Error` for anything that is imminently recoverable by the caller.

Reply:
It's a little bit more admissive of some inconsistencies this way -- the caller is interested in knowing whether or not it is safe to continue operating if it switches to `bhh` as the current block hash. That may not be true for a variety of reasons -- if errors are raised by any of the functions called by this function, that's the case.

However, to return an accurate boolean would require a bunch of checks of the thrown errors (which may not even be convey enough information to determine whether or not a NotFoundError meant that `bhh` wasn't in the current fork, or the MARF was otherwise corrupted), so I'm hesitant to commit to that interface. I think the thing to do is probably rename this function back to `check_is_ancestor_hash`.

I think the current interface of `MARF::get` gets this wrong --- various codepaths may throw some errors if a key doesn't exist in the MARF, but those aren't encoded get in the `Option` return of the function.

Reply:
> I think the current interface of MARF::get gets this wrong --- various codepaths may throw some errors if a key doesn't exist in the MARF, but those aren't encoded get in the Option return of the function.

Sorry, I had a hard time parsing this sentence.  Can you elaborate a bit more?

Reply:
If a key isn’t in a MARF, get can return either:

Err(NotFoundError)

or

Ok(None)

Reply:
That's what I thought you were saying.  That's a problem.  It should be returning Ok(None) if the path wasn't found, and Err(...) for any other reason.

Reply:
I think it's straightforward to fix though?  Just map Err(NotFoundError) to Ok(None) in the call to MARF::get_path() in MARF::get().

Reply:
Yep -- will update `MARF::get()` and change the name of `is_ancestor_hash` to `check_ancestor_hash`.

Reply:
Looks good!